database/database.go

```diff
@@ -207,11 +208,11 @@ func Open(ctx context.Context, shutdownWg *sync.WaitGroup, dbFile string, backup
 		for {
 			select {
 			case <-time.After(backupInterval):
-				err := writeHotBackupFile(db)
+				err := vdb.writeHotBackupFile()
 				if err != nil {
 					log.Errorf("Failed to write database backup: %v", err)
 				}
-			case <-ctx.Done():
+			case <-shutdownCtx.Done():
 				shutdownWg.Done()
```

Comment:
I think it's very important to call `db.Close()` here before `shutdownWg.Done()`. I think the entire reason for having the waitgroup. We want to be sure the db is done writing before we shut down. If we cut it off mid write, that could lead to a corrupted db.

Reply:
OK! I see it now in `vspd.go`. Sorry I missed that. I think you want to still do `db.Close()` for the two possible errors in this function, when it would return with error and then that `.Close` will not happen in `vspd.go`

Reply:
Good catch. I've pushed another commit for this.